gnd/src/main/java/com/google/android/gnd/ui/home/HomeScreenFragment.java

```diff
@@ -332,6 +365,54 @@ private void showOfflineAreas() {
     viewModel.showOfflineAreas();
   }
 
+  private void onShowAddPolygonDialogRequest(Point point) {
+    if (vertices.contains(point)) {
```

Comment:
`contains` will required `point` to be exactly equal to one of the vertices. This is nearly impossible in practice; the coordinates are floats, so they may be close but not exactly equal. Instead, we need to match with some degree of tolerance.

Also, this will match _any_ vertex, but a polygon is only complete when the last vertex ~= the first vertex.